core/src/main/scala/zio/config/PropertyTree.scala

```diff
@@ -75,16 +77,18 @@ sealed trait PropertyTree[+K, +V] { self =>
   final def getOrElse[K1 >: K, V1 >: V](tree: => PropertyTree[K1, V1]): PropertyTree[K1, V1] =
     if (self == PropertyTree.empty) tree else self
 
+  @tailrec
   final def getPath[K1 >: K](k: List[K1]): PropertyTree[K1, V] =
     k match {
       case Nil => self
       case head :: next =>
         self match {
-          case Empty         => Empty
-          case Leaf(_)       => Empty
-          case Record(value) => value.get(head.asInstanceOf[K]).map(_.getPath(next)).getOrElse(Empty)
-          case Sequence(r)   => Sequence(r.map(_.getPath(k)))
-
+          case Empty | Leaf(_) | Sequence(_) => Empty
```

Comment:
This is a big change in functionality. I think I like it but it does have lots of downstream implications...

Reply:
Actually no. There is only 1 use case of this method.

Reply:
Yes, but that method is the whole basis of `read`. Anyway, if the tests pass, I'm happy with it!

Reply:
That the point: `read` is reimplemented.